bin/migrate-oats-data/noi/noi_decisions/notice_of_intent_decisions_init.py

```diff
@@ -0,0 +1,191 @@
+                    %(rescinded_comment)s,
+                    %(rescinded_date)s,
+                    %(resolution_number)s,
+                    %(resolution_year)s,
+                    true,
+                    %(notice_of_intent_uuid)s,
+                    %(is_subject_to_conditions)s,
+                    %(oats_alr_appl_decision_id)s
+                )
+                ON CONFLICT ("resolution_number", "resolution_year")
+                    WHERE "audit_deleted_date_at" is null and "resolution_number" is not NULL
+                DO UPDATE SET 
+                date = COALESCE(EXCLUDED.date, alcs.notice_of_intent_decision.date),
+                decision_maker = COALESCE(EXCLUDED.decision_maker, alcs.notice_of_intent_decision.decision_maker),
+                outcome_code = COALESCE(EXCLUDED.outcome_code, alcs.notice_of_intent_decision.outcome_code);
+    """
+    return query
+
+
+def _prepare_oats_alr_applications_data(row_data_list):
+    data_list = []
+    for row in row_data_list:
+        resolution_number, year = _map_resolution_number_and_year(row)
+        mapped_row = {
+            "audit_created_by": OATS_ETL_USER,
+            "created_at": add_timezone_and_keep_date_part(row.get("when_created")),
+            "date": add_timezone_and_keep_date_part(row.get("decision_date")),
+            "decision_description": row.get("decision_desc"),
+            "decision_maker": "CEO",
+            "is_draft": False,
+            "outcome_code": _map_outcome_code(row),
+            "rescinded_comment": row.get("rescinded_comments"),
+            "rescinded_date": add_timezone_and_keep_date_part(
+                row.get("rescinded_date")
+            ),
+            "resolution_number": resolution_number,
+            "resolution_year": year,
+            "was_released": True,
+            "notice_of_intent_uuid": row.get("noi_uuid"),
+            "is_subject_to_conditions": _map_is_subject_to_conditions(row),
+            "oats_alr_appl_decision_id": row.get("alr_appl_decision_id"),
+        }
+        data_list.append(mapped_row)
+
+    return data_list
+
+
+def _map_outcome_code(row):
+    if row.get("outright_refusal_ind", None) == "N":
+        return "APPR"
+    elif row.get("outright_refusal_ind", None) == "Y":
+        return "ONTP"
+    return None
+
+
+def _map_resolution_number_and_year(row):
+    year_and_number = str(row["resolution_number"])
+    resolution_number = int(year_and_number[:-4])
+    year = int(year_and_number[-4:])
+    return resolution_number, year
+
+
+def _map_is_subject_to_conditions(row):
+    return row.get("is_subject_to_conditions", "False") == True
+
+
+@inject_conn_pool
+def clean_notice_of_intents(conn=None):
+    logger.info("Start notice_of_intent_decisions cleaning")
+    with conn.cursor() as cursor:
+        cursor.execute(
+            f"DELETE FROM alcs.notice_of_intent_decision WHERE audit_created_by = '{OATS_ETL_USER}' AND audit_updated_by = null"
```

Comment:
'= null' does not seem to function as intended in previous ETLs consider changing to IS NULL